js/hitbtc3.js

```diff
@@ -1746,6 +1748,78 @@ module.exports = class hitbtc3 extends Exchange {
         return this.filterBySymbolSinceLimit (sorted, symbol, since, limit);
     }
 
+    async fetchFundingRate (symbol, params = {}) {
+        await this.loadMarkets ();
+        const market = this.market (symbol);
+        if (!market['swap']) {
+            throw new BadSymbol (this.id + ' fetchFundingRate() supports swap contracts only');
+        }
+        const request = {};
+        if (symbol !== undefined) {
+            symbol = market['symbol'];
+            request['symbols'] = market['id'];
+        }
+        const response = await this.publicGetPublicFuturesInfo (this.extend (request, params));
+        //
+        //     {
+        //         "BTCUSDT_PERP": {
+        //             "contract_type": "perpetual",
+        //             "mark_price": "42307.43",
+        //             "index_price": "42303.27",
+        //             "funding_rate": "0.0001",
+        //             "open_interest": "30.9826",
+        //             "next_funding_time": "2022-03-22T16:00:00.000Z",
+        //             "indicative_funding_rate": "0.0001",
+        //             "premium_index": "0",
+        //             "avg_premium_index": "0.000029587712038098",
+        //             "interest_rate": "0.0001",
+        //             "timestamp": "2022-03-22T08:08:26.687Z"
+        //         }
+        //     }
+        //
+        const data = this.safeValue (response, market['id']);
+        return this.parseFundingRate (data, market);
+    }
+
+    parseFundingRate (contract, market = undefined) {
+        //
+        //     {
+        //         "contract_type": "perpetual",
+        //         "mark_price": "42307.43",
+        //         "index_price": "42303.27",
+        //         "funding_rate": "0.0001",
+        //         "open_interest": "30.9826",
+        //         "next_funding_time": "2022-03-22T16:00:00.000Z",
+        //         "indicative_funding_rate": "0.0001",
+        //         "premium_index": "0",
+        //         "avg_premium_index": "0.000029587712038098",
+        //         "interest_rate": "0.0001",
+        //         "timestamp": "2022-03-22T08:08:26.687Z"
+        //     }
+        //
+        const nextFundingDatetime = this.safeString (contract, 'next_funding_time');
+        const datetime = this.safeString (contract, 'timestamp');
+        return {
+            'info': contract,
+            'symbol': this.safeSymbol (market['id'], market),
```

Comment:
We should not use the market['id'] in the first argument to safeSymbol, instead we should use the id replied by the exchange in its response, if available. In this case it's not available, so we use `undefined`.